server/src/test/java/org/opensearch/action/support/replication/TransportReplicationActionTests.java

```diff
@@ -950,7 +952,10 @@ public void testSeqNoIsSetOnPrimary() {
         Set<String> inSyncIds = randomBoolean()
             ? singleton(routingEntry.allocationId().getId())
             : clusterService.state().metadata().index(index).inSyncAllocationIds(0);
-        ReplicationGroup replicationGroup = new ReplicationGroup(shardRoutingTable, inSyncIds, shardRoutingTable.getAllAllocationIds(), 0);
+        Set<String> trackedAllocationIds = shardRoutingTable.getAllAllocationIds();
+        Map<String, ReplicationTracker.ReplicationMode> replicationModeMap = trackedAllocationIds.stream()
+            .collect(Collectors.toMap(v -> v, v -> ReplicationTracker.ReplicationMode.FULL_REPLICATION));
```

Comment:
Sorry but finding it hard to understand how `replicationModeMap ` with values `ReplicationMode.FULL_REPLICATION` and then later `ReplicationModeAwareShardRouting ` is trying to establish.

Reply:
So, in a replication group we have all shards (primary and replicas) and we calculate the replication group using the logic mentioned in calculateReplicationGroup method. While calling the individual shards in performOnReplica method in ReplicationOperation, it runs replication request on all shards except the primary. In existing local translog world, all the shards would be replicated and thus have the attribute that they are qualified to be replicated. 

Now consider a remote store world, where we have multiple shards (primary, relocating primary, replicas). In this case, we would want the request be replicated on primary, relocating primary shard, but not on replicas. So, the need for replication has been defined as enum ReplicationMode (which is per shard). Now, similarly, the replicationTarget gets calculation which holds the ReplicationModeAwareShardRouting. Individual ReplicationProxy can decide what they want to do with this information. In case of remote translog world, we want all TransportReplicationAction (except TransportShardBulkAction and PublishCheckpointAction) to disallow replication requests to fan out to shards that do not have replicationMode as FULL_REPLICATION. However, we still want the calls to fan out to replicationMode FULL_REPLICATION. Hence, the replicationMode is mapped to per shard.

Reply:
hope this answer your doubt. @Bukhtawar 

In this particular example, all tracked allocation ids are marked as FULL_REPLICATION as this test handles the existing test case where remote store is not enabled, but the interface for calculateReplicationGroup has changed. To handle change in interface and test to run as status quo, the change has been done in this particular test case.

Reply:
As discussed offline, we might not need `ReplicationModeAwareShardRouting ` at all, given it gives no indication of what the source is. Just having target with replication mode sounds incomplete

Reply:
@Bukhtawar incorporated.